tests/test_ops/test_load_ontology.py

```diff
@@ -0,0 +1,135 @@
+    result = load_ontology(op_context)
+    
+    # Verify the correct parameters were used to initialize OntologyLoaderController
+    mock_ontology_loader.assert_called_once_with(
+        source_ontology="envo",
+        output_directory=op_context.op_config["output_directory"],
+        generate_reports=False
+    )
+    
+    # Verify that run_ontology_loader was called
+    mock_instance.run_ontology_loader.assert_called_once()
+    
+    # The function doesn't have a return value
+    assert result is None
+
+
+
+@pytest.mark.skipif(
+    os.getenv("MONGO_PASSWORD") is None or os.getenv("ENABLE_DB_TESTS") != "true",
+    reason="Skipping test: Requires MONGO_PASSWORD and ENABLE_DB_TESTS=true",
+)
+def test_load_ontology_integration(op_context):
+    """Tests the load_ontology op with actual MongoDB connection and verifies results"""
+    try:
+        # Get MongoDB client
+        mdb = op_context.resources.mongo.db
+        
+        # Print detail about the MongoDB connection
+        print(f"Connected to MongoDB: {op_context.resources.mongo.db.client.address}")
+        
+        # Check if ontology collections exist before running
+        ontology_class_set_before = mdb.get_collection("ontology_class_set").count_documents({})
+        ontology_relation_set_before = mdb.get_collection("ontology_relation_set").count_documents({})
+        
+        print(f"Before running: {ontology_class_set_before} classes, {ontology_relation_set_before} relations")
+        
+        # Execute the op
+        result = load_ontology(op_context)
+
+        # Verify the op behavior:
+        # 1. Check that ontology_class_set has entries
+        ontology_class_count = mdb.get_collection("ontology_class_set").count_documents({})
+        print(f"After running: {ontology_class_count} classes")
+
+        # 2. Check that ontology_relation_set has entries
+        ontology_relation_count = mdb.get_collection("ontology_relation_set").count_documents({})
+        print(f"After running: {ontology_relation_count} relations")
+        
+        # 3. Check for some known ENVO terms if we have ontology data
+        assert ontology_class_count > 0
+        assert ontology_relation_count > 0
+
+        sample_envo_id = "ENVO:00000001"  # Example ENVO ID
+        envo_term = mdb.get_collection("ontology_class_set").find_one({"id": sample_envo_id})
+        assert envo_term is not None
+        
+        # 4. Check report files if generate_reports was True
+        if op_context.op_config["generate_reports"]:
+            output_dir = op_context.op_config["output_directory"]
+            assert os.path.exists(os.path.join(output_dir, "ontology_inserts.tsv"))
+            assert os.path.exists(os.path.join(output_dir, "ontology_updates.tsv"))
+        
+        # 5. Verify the function has no return value (was incorrectly expected to be 0)
+        assert result is None
+    except Exception as e:
+        print(f"An error occurred: {e}")
+        raise
```

Comment:
I don't think this try/except is really doing anything. Pytest tells you if an uncaught exception is raised.